### PR TITLE
distributed: limit size of created partitions

### PR DESCRIPTION
**Pull request by matthiasdiener**
This does not work yet since it causes a PartitionInducedCycleError.

Some statistics for y2-isolator:
- total number of nodes in DAG (unpartitioned): 6728

### CODE REVIEW

File: pytato/equality.py [hunk 1]
```diff
@@ -231,7 +231,8 @@ def map_loopy_call(self, expr1: LoopyCall, expr2: Any) -> bool:
                 )
 
     def map_loopy_call_result(self, expr1: LoopyCallResult, expr2: Any) -> bool:
-        return (self.rec(expr1._container, expr2._container)
+        return (expr1.__class__ is expr2.__class__
+                and self.rec(expr1._container, expr2._container)
```
Comment (matthiasdiener) [inline pytato/equality.py:235]: @kaushikcfd - Does this change look reasonable? If so, I could open a separate PR with just this change (expr2 does not always have a `_container` attribute)

Comment (kaushikcfd) [reply pytato/equality.py:235]: Yep, that is very reasonable. Thanks for fixing this! (A different PR would be great.)

Looks like we also missed this condition in `map_named_array`.

Comment (matthiasdiener) [reply pytato/equality.py:235]: Done in #250.

**Comment by matthiasdiener** [comment]
Closing as unnecessary

**Pull request closed without merging**